ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -254,7 +256,7 @@ export default class ConfirmTransactionBase extends Component {
               onHeaderClick={() => this.handleEditGas()}
               secondaryText={hideFiatConversion ? this.context.t('noConversionRateAvailable') : ''}
             />
-            {advancedInlineGasShown
+            {advancedInlineGasShown || (!isMainnet && !process.env.IN_TEST)
```

Comment:
The `ConfirmDetailRow` component above is where the "Edit" button is that we need to hide.

It looks like the `ConfirmDetailRow` component expects the `headerText` prop to always be passed in. I'd suggest making that prop optional, and omitting all three related props (`headerText`, `headerTextClassName`, and `onHeaderClick`) when on testnets.